### PR TITLE
Improved support for Vive Trackers on biceps

### PR DESCRIPTION
**Pull request by hyperlogic**
Improved support for Vive Trackers on biceps.  Two new configuration fields appear on the Controller Settings page.
   * Arm Circumference - Measurement of the bicep circumference, around which the puck is attached.
   * Shoulder Width - If the user has his hands at his sides, this is the measurement of the distance between the two biceps, as measured from the humerus bone.  See image below

shoulder width

Other misc items:
   * Removed arm targets from avatar-animation.json
   * Cleaned up copying of poses from input system into avatar.  Now a single function is used MyAvatar::setControllerPoseInSensorFrame(), instead of one per set of joints.
   * AnimInverseKinematic node now has list of secondary targets in addition to primary targets.  These secondary targets are applied to the initial solution/underPose before solving.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1978/HighFidelity-Beta-PR11053-572213daaf6ac3c4a90c87f51d95ab82a63614cf.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows2016/2026/HighFidelity-Beta-PR11053-2a45c6d3dc170b913cd2c5519c97de4ea2f70aae.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows2016/2034/HighFidelity-Beta-PR11053-78c6564174b35fa7a330e2d3874a83a18674d113.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2038/HighFidelity-Beta-PR11053-bcb4cc32d26718ee9c622202cc7339252dc44e87.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows2016/2038/HighFidelity-Beta-PR11053-bcb4cc32d26718ee9c622202cc7339252dc44e87.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2072/HighFidelity-Beta-PR11053-6c2b6674bb22336a83d95b8830f51248b82b004b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2072/HighFidelity-Beta-PR11053-6c2b6674bb22336a83d95b8830f51248b82b004b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2149/HighFidelity-Beta-PR11053-2383933252278d7c82bc7e5260c8d896c3bcb92b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2149/HighFidelity-Beta-PR11053-2383933252278d7c82bc7e5260c8d896c3bcb92b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2183/HighFidelity-Beta-PR11053-25b5cb4762a0969609bfcc9ce474edabaf4f1d02.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2183/HighFidelity-Beta-PR11053-25b5cb4762a0969609bfcc9ce474edabaf4f1d02.exe

**Comment by hyperlogic** [comment]
The following issues with this PR have been fixed:
  * elbows locking during flex-biceps-like-macho-man recording.
  * hand not reaching primary target during shoulders-freak-out recordings.

However some issues were not fixed, but were looked into. These issues will be helped by moving the arm pucks downward, more toward the elbow.
  * elbows are too far backward during flex-biceps-like-macho-man recording.
  * elbows do not touch during elbows-touch-together recordings

Finally, the popping during the chicken-dance recordings are due to lost tracking, causing the pucks to freeze in place momentarily.

**Comment by hyperlogic** [comment]
QA Test plan:
  * Wear feet, hips and arm pucks.  Make sure arm pucks are low on your bicep as close to your elbows as possible.
  * Enter Settings > Controller Settings, select the shoulder configuration.  Enter your measured arm circumference and shoulder width parameters.  Press calibrate and enter the t-pose.
  * Your avatar's elbows should now follow your physical elbows.
  * Additionally: use the inputRecorder script to record the following poses
    * roll shoulders forward
    * roll shoulders backward
    * open and close chest
    * bench press
    * flex biceps like a macho man
    * field goal
    * reverse field goal
    * touch elbows together
    * chicken dance
  * At no point should the hand fail to reach your hand controller.

**Comment by hyperlogic** [comment]
https://highfidelity.fogbugz.com/f/cases/5176/Fix-Shoulder-Calibration-for-Users-with-Different-Limb-Lenghts-than-Avatar

**Comment by anshuman64** [comment]
Tony, this is a HUGE improvement from before. Great work!

Here's a video: https://drive.google.com/open?id=0B_nmInIG19-uZ1FCcTRiRDZaUm8

Two items we should address in future work:
1. That JimJamz bug: https://highfidelity.fogbugz.com/f/cases/6590/Pole-Vectors-Elbow-Bug-JimJamz-video-attached
2. It is too difficult to calibrate with your palms down as expected. We should do the same thing as the feet where we tell users exactly where to keep the puck, and allow users to calibrate elbows facing up as natural with Vive controllers. https://highfidelity.fogbugz.com/f/cases/6749/Calculate-proper-offset-given-fixed-position-for-shoulder-pucks

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2274/HighFidelity-Beta-PR11053-98bbe887b9d6efc6b52ecff96bbf154df426264a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2274/HighFidelity-Beta-PR11053-98bbe887b9d6efc6b52ecff96bbf154df426264a.exe

File: libraries/shared/src/GeometryUtil.cpp [hunk 1]
```diff
@@ -605,3 +605,55 @@ float coneSphereAngle(const glm::vec3& coneCenter, const glm::vec3& coneDirectio
 
     return glm::max(0.0f, theta - phi);
 }
+
+// given a set of points, compute a best fit plane that passes as close as possible through all the points.
+// http://www.ilikebigbits.com/blog/2015/3/2/plane-from-points
+bool findPlaneFromPoints(const glm::vec3* points, size_t numPoints, glm::vec3& planeNormalOut, glm::vec3& pointOnPlaneOut) {
```
Comment (sethalves) [inline libraries/shared/src/GeometryUtil.cpp]: Do you have plants for this?  I don't see it used.

**Pull request merged**